openhcl/openhcl_boot/src/main.rs

```diff
@@ -605,13 +605,17 @@ fn shim_main(shim_params_raw_offset: isize) -> ! {
         )
         .vtl0_alias_map_available()
     {
-        // Disable the alias map on ARM because physical address size is not
-        // reliably reported. Since the position of the alias map bit is inferred
-        // from address size, the alias map is broken when the PA size is wrong.
-        // TODO: is this still true?
+        // On ARM, use the physical address width reported via device tree since
+        // it is not reliably reported architecturally. Since the position of
+        // the alias map bit is inferred from address size, the alias map is
+        // broken when the PA size is wrong.
         if !cfg!(target_arch = "aarch64") {
```

Comment:
On x86, can we use devicetree when it's available and skip the architectural query, rather than do this validation? This gives the host more flexibility to move things around in the future.